design/007-provider-versioning-strategy.md

```diff
@@ -0,0 +1,54 @@
+# Provider Separation on specific CRDs
+```yaml
+---
+title: Provider Separation on specific CRDs
+version: v1alpha1
+authors: Gustavo Carvalho
+creation-date: 2023-08-25
+status: draft
+---
+```
+
+WRT: https://github.com/external-secrets/external-secrets/issues/694
+
+We want to separate provider configuration from the SecretStore, in a way that allows us to install providers only when needed. This also allows us to version provider fields accordingly to their maturity without impacting the SecretStore Manifest
+
+### Proposed Changes
+The changes to the code proposed are summarized below:
+* Add a new CRD group called `providers` where all provider configuration will reside as individual CRD.
+* Add a new Field `ProviderRef` to the SecretStore manifest.
```

Comment:
What's the point of having the SecretStore? I don't think it provides enough functionality to justify its existence. Can't we just drop the SecretStore and instead reference a `Provider` from the `ExternalSecret` directly?

The secret store provides:
* `controller`
* `refreshInterval`
* `retrySettings`
* `provider config`

And that's it. The first 3 can be implemented with a `GenericProvider` interface that every provider CRD must implement, the rest is provider-specific.

Reply:
We can, I was just thinking in a way that didn’t cause breaking changes (your second comment goes on that direction, so I’ll take a look how feasible it is)

Reply:
https://github.com/external-secrets/external-secrets/pull/2655#discussion_r1391144828